applications/api.py

```diff
@@ -57,7 +57,16 @@ def derive_application_state(bootcamp_application):  # pylint: disable=too-many-
         return AppStates.AWAITING_SUBMISSION_REVIEW.value
     elif len(submissions) < bootcamp_application.bootcamp_run.application_steps.count():
         return AppStates.AWAITING_USER_SUBMISSIONS.value
-    if bootcamp_application.order is None or bootcamp_application.order.status != Order.FULFILLED:
+    total_paid = get_total_paid(
+        bootcamp_application.user,
+        bootcamp_application.bootcamp_run.run_key,
+        application_id=bootcamp_application.id,
+    )
+    if (
+        total_paid is None or
```

Comment:
`total_paid` should never be `None`. If there are no payments made it returns a `Decimal(0)` instead